osquery/tables/networking/darwin/wifi.mm

```diff
@@ -39,11 +39,41 @@
     {"disabled", "Disabled"},
     {"temporarily_disabled", "TemporarilyDisabled"}};
 
-// Check if we are running on OS X 10.9, where the key in plist is different
+// The name of the "last_connected" key changed in 10.15, and the
+// "auto_login" key no longer exists.
+const std::map<std::string, std::string> kKnownWifiNetworkKeys = {
+    {"ssid", "SSID"},
+    {"network_name", "SSIDString"},
+    {"security_type", "SecurityType"},
+    {"roaming_profile", "RoamingProfileType"},
+    {"auto_login", "AutoLogin"}, // actually no longer present since Catalina
+    {"last_connected", "LastAutoJoinAt"},
```

Comment:
Since this is the only unique key change, I recommend having three maps, a `kKnownWifiNetworkKeys`, then a pre and post catalina map. Below you can loop over the shared set of keys and then loop over the version-specific set of keys. I would like to prevent as much duplication as possible.

Reply:
+1 to less duplication. Depending on how you predict the future, the other pattern is to use an if block to append items building it.